components/StoreLocatorModal.vue

```diff
@@ -0,0 +1,222 @@
+                    <span class="sf-property__name">{{ hours.day }}</span>
+                    <span class="sf-property__value">
+                      {{ hours.openTime }} - {{ hours.closeTime }}
+                    </span>
+                  </div>
+                </div>
+              </SfAccordionItem>
+            </SfAccordion>
+          </template>
+        </SfRadio>
+        <div class="section-border"></div>
+      </div>
+    </div>
+  </SfModal>
+</template>
+<script lang="ts">
+import { SfModal, SfRadio, SfSearchBar, SfAccordion, SfChevron, SfButton } from "@storefront-ui/vue"
+
+import { useAsync } from "@nuxtjs/composition-api"
+import useUiState from "../composables/useUiState"
+import { useCurrentLocation } from "../composables/storeFinder/useCurrentLocation"
+import { useStoreLocations } from "../composables/storeFinder/useStoreLocations"
+import { usePurchaseLocation } from "../composables/storeFinder/usePurchaseLocation"
+
+import { storeLocationGetters } from "@/composables/getters"
+
+export default {
+  name: "LoginModal",
+  components: {
+    SfModal,
+    SfRadio,
+    SfSearchBar,
+    SfAccordion,
+    SfChevron,
+    SfButton,
+  },
+  setup() {
+    const { isStoreLocatorOpen, toggleStoreLocatorModal } = useUiState()
+    const { currentLocation, loadWithNavigator } = useCurrentLocation()
+    const { locations, search: searchStoreLocations } = useStoreLocations()
+    const { set, load: loadPurchaseLocation } = usePurchaseLocation()
+    const selectedStore = ref("")
+
+    useAsync(async () => {
+      await searchStoreLocations()
+    }, null)
+
+    const handleCurrentLocation = async () => {
+      await loadWithNavigator()
+      await searchStoreLocations({
+        latitude: currentLocation.value.latitude,
+        longitude: currentLocation.value.longitude,
+      })
+    }
+
+    const closeModal = () => {
+      toggleStoreLocatorModal()
+      selectedStore.value = ""
+    }
+
+    const storeDetails = computed(() => {
+      return storeLocationGetters.getLocations(locations.value)
+    })
+
+    const handleStoreChange = (locationCode: string) => {
+      selectedStore.value = locationCode
+      set(locationCode)
+      loadPurchaseLocation()
+    }
+
+    return {
+      currentLocation,
+      isStoreLocatorOpen,
+      locations,
+      storeDetails,
+      selectedStore,
+      handleStoreChange,
+      handleCurrentLocation,
+      toggleStoreLocatorModal,
+      closeModal,
+    }
+  },
+}
+</script>
+<style lang="scss" scoped>
+@import "~@storefront-ui/vue/styles";
+.sf-modal {
+  --modal-width: 39.375rem;
+  --modal-content-padding: 0;
+}
+.title {
+  font-weight: 700;
+  font-size: var(--h4-font-size);
+  padding: var(--spacer-2xs) var(--spacer-lg);
+}
+.location-search {
+  display: flex;
+  justify-content: space-between;
+  align-items: stretch;
+  padding: var(--spacer-lg) var(--spacer-base) var(--spacer-2xs) var(--spacer-lg);
+}
+.sf-button__small {
```

Comment:
I think as per the convention it should be .sf-button--small instead of .sf-button__small